### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this change provide BC-break.

## Changelog

```markdown
### Changed
- Change using `Sonata\Doctrine\Model\PageableManagerInterface` in favor of `Sonata\DatagridBundle\Pager\PageableInterface`
### Removed
- Removed support for `aws/aws-sdk-php` < 3.0
- Removed support for `sonata-project/datagrid-bundle` < 3.0
```

### CODE REVIEW

**Comment by wbloszyk** [comment]
@sonata-project/contributors 
Any idea why `\SoapClient` is missing on both php 7.3 tests? 
I try `"ext-soap": "*",` but it did not work. 😞

**Comment by jordisala1991** [comment]
Did you tried to add it to github actions, on the setup php extensions

**Comment by wbloszyk** [comment]
> Did you tried to add it to github actions, on the setup php extensions

No, I did not. It is working on `3.x` branch and should work here too. Also I rerun green tests and they red now - https://github.com/sonata-project/SonataMediaBundle/pull/1916. Tests pass but self deprecations make it red.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
                   php-version: ${{ matrix.php-version }}
                   coverage: pcov
                   tools: composer:v2, pecl
-                  extensions: mongodb-1.9.0, soap
+                  extensions: mongodb-1.9.0, soap, gd2
```
Comment (VincentLanglet) [inline .github/workflows/test.yaml]: Dont forget you'll need a PR on dev-kit when it'll work.

Comment (wbloszyk) [reply .github/workflows/test.yaml]: I just working on it.

Comment (wbloszyk) [reply .github/workflows/test.yaml]: Can you look at https://github.com/sonata-project/dev-kit/pull/1450

**Comment by VincentLanglet** [comment]
Done in https://github.com/sonata-project/SonataMediaBundle/pull/1965

**Pull request closed without merging**